### PR TITLE
fix(query): Prevent multiple entries for same predicate/type in schema(#7715)

### PR DESCRIPTION
**Pull request by dshekhar95**
…a mutations. (#7715)

* fix(query): Prevent multiple entries for the same predicate/type in mutations.

Fixes DGRAPH-3225

## Problem
multiple entries for the same predicate/type in mutations.

 

## Solution
Add a check before adding predicate/type to check if it already exists

### CODE REVIEW

**Comment by dshekhar95** [comment]
@skrdgraph @joshua-goldstein @meghalims Tests passing. Can we approve to merge this cherry-pick?

**Comment by skrdgraph** [comment]
@dshekhar95 could you fix the PR description

**Comment by skrdgraph** [comment]
Now that we have `coverage` in place :) .. I am very nervous to merge this PR. Lets discuss this in cherry-pick meeting. Parallely let me look to understand further on what it is doing.

**Comment by dshekhar95** [comment]
> @dshekhar95 could you fix the PR description

done

**Comment by dshekhar95** [comment]
> Now that we have `coverage` in place :) .. I am very nervous to merge this PR. Lets discuss this in cherry-pick meeting. Parallely let me look to understand further on what it is doing. Give me a little time to review & test this.

yea that looks worrisome....but that would lead us to the conclusion that we shouldnt be doing any cherry picks until we do that?

**Review by dshekhar95 — COMMENTED** [review]
@rarvikar We have added a test case to go along with this Cherry-pick. Can you please advise how we might add a test case to check for duplicate custom types (ex. a Person) given the format of the schemas? See lines 149-165 for the 3 current tests.

**Comment by kevinmingtarja** [comment]
@dshekhar95 I have just added one test case for duplicated types.

cc: @rarvikar

**Comment by dshekhar95** [comment]
> @dshekhar95 I have just added one test case for duplicated types.
> 
> cc: @rarvikar

Looks good! Rahul: looks like Kevin figured it out

File: edgraph/server.go [hunk 1]
```diff
@@ -287,7 +287,15 @@ func parseSchemaFromAlterOperation(ctx context.Context, op *api.Operation) (*sch
 		return nil, err
 	}
 
+	preds := make(map[string]struct{})
```
Comment (matthewmcneely) [inline edgraph/server.go:290]: Note to reviewers mystified by this, `map[string]struct{}` is poor-man's _set_ in golang

**Comment by matthewmcneely** [comment]
Note to reviewers, seems there was some issue with rebasing this branch, the relevant changes are in edgraph/server.go and edgraph/server_test.go

**Review by matthewmcneely — APPROVED** [review]
